SConstruct

```diff
@@ -1333,10 +1333,14 @@ sanitizers = env.ParseList(GetOption('sanitizers'), supported_sanitizers)
 if sanitizers:
     if not compiler in ['gcc', 'clang']:
         env.Die("sanitizers are not supported for compiler '%s'" % compiler)
+    
+    if compiler=='clang':
+        env.AppendUnique(CXXFLAGS=['-fsanitize-blacklist=src/sanitizer_blacklist.txt'])
```

Comment:
It would be also nice to make `src/sanitizer_blacklist.txt` the dependency of all source or object files, so that they are rebuilt when this file changes.

Here is an example of making a file the dependency of .so file: https://github.com/roc-streaming/roc-toolkit/blob/develop/src/SConscript#L56

One way to do it is to iterate all sources and/or objects arrays in SConscript and explicitly call [Depends()](https://www.scons.org/doc/0.96.91/HTML/scons-user/x995.html) for each file. I'm not sure whether we may call it for sources, probably it will work properly only for objects. In this case we'll probably need to find all places when sources are passed to builders like SharedLibrary() and Program() directly, and instead explicitly construct Object() for each source file, call Depends() for each object, and then pass objects array instead of sources array to builders.

Another way is to write a scanner that will produce the needed dependency: https://scons.org/doc/1.2.0/HTML/scons-user/c3755.html. This solution should be much more compact but I didn't try it before. I suggest to try it first.

Reply:
Maybe we have to use first one.

Cause there seems a bug with `Scanner` factory function.
[See this](https://stackoverflow.com/q/62946778/8272206)

Reply:
Interesting, did you see the recent edit to the answer?

Reply:
Yes I just checked it out. Now it is fine.

Reply:
I have one doubt.

When I use following code to specify dependencies in `SConstruct` it works as expected(after update to blacklist file cmdline.cpp gets recompiled).
`env.Depends('#build/x86_64-pc-linux-gnu/clang-6.0.0-debug/tools/roc_recv/cmdline.o', '#src/sanitizer_blacklist.txt')`

But When I use the following codes instead to specify that all objects depend on blacklist file it does not work as expected. In particular, after updating the blacklist file `cmdline.o` does not get rebuilt. Instead, it says everything is up to date.

`
dlist = env.GlobRecursive(".", "*.o")
env.Depends(env.File(dlist), '#src/sanitizer_blacklist.txt')
`

I have temporarily disabled implicit cache.

Reply:
Are you sure dlist contains expected object files? IIRC GlobRecursive searches in source directory, not in build directory.

In general, just searching for all .o files feels like a bad idea. For example, corresponding . o file may not be created on filesystem yet if this is the first run after clean. Thus, the dependency on blacklist will be added only at second scons invocation. And changing dependencies may cause unwanted rebuilds of object files. Also, object file naming is platform specific.

So instead of searching .o files, I'd either collect scons targets returned by Object() builder, or use scanner if possible.

Reply:
Yes, that really is not a good way. I was just trying to get familiar with scons.

Reply:
Just checked: yes, GlobRecursive searches in source dir: https://github.com/roc-streaming/roc-toolkit/blob/develop/site_scons/site_tools/roc/fs.py#L20 (because of srcnode())

Reply:
But I printed `dlist` and output contained `build/x86_64-pc-linux-gnu/clang-6.0.0-release/tools/roc_send/cmdline.o` among many other object files.

Reply:
No idea :) Maybe try to iterate over dlist and call File() and Depends() for each entry individually. Also you can check that the File() object really refers to the correct file by getting it's abspath property. Sometimes the relative path looks good, but it's relative to a wrong base. You can also try `scons ... --debug=explain`